### DISCUSSION TITLE
How to connect to an SFTP server?

### DISCUSSION THREAD

**Discussion by dylankenneally** [dylankenneally/react-native-ssh-sftp#17] [Q&A] [answered]
Asking the question on behalf of @aymanMFD, who raised the this topic as an issue, [here](https://github.com/dylankenneally/react-native-ssh-sftp/issues/16).

> Hi!
I am new to React and its libraries and I can't seem to figure out how to connect to an SFTP server. Its true that in the README file there are the steps and all, but since I am a beginner I cant seem to understand, sorry for the inconvenience.
Hope you can help

**Comment by dylankenneally** [comment]
Happy to try help you get started @aymanMFD.

Do you have an example repo that I can see to help get an idea of where you are at? Or specific questions about the README content to help you get started?

**Comment by aymanMFD** [comment]
I apologize in advance im still new to github and all.
But all i really need is how do I connect to an SFTP server.
You made it clear on how to do it to an SSH but idk how to do it for an SFTP.

**Comment by dylankenneally** [comment, marked as answer]
This library does support SFTP for both Android & iOS, check out [this section](https://github.com/dylankenneally/react-native-ssh-sftp?tab=readme-ov-file#sftp) in the readme for API details.

To use SFTP, you need an SSH connection, as SFTP runs over the SSH protocol as a subsystem. So you'll want something like the following:

```jsx
const server = 'your ssh server address';
const sshUsername = 'your ssh user name, maybe root';
const sshPwd = 'your ssh password';

// get your SSH connection going first, note this is the SSH server running your SFTP software
const client = await SSHClient.connectWithPassword(server, 22, sshUsername, sshPwd);

// now we connect to the SFTP server
await client.connectSFTP();

// and do stuff with that
const path = '.';
const response = client.sftpLs(path);
console.log(response);

// and clean up when you're done
client.disconnectSFTP(); // close the SFTP connection, not 100% needed, but good practice
client.disconnect(); // close the SSH connection to the server
```

Does that help you get started?

**Reply by aymanMFD** [reply]
Ohh sweet thats exactly what i was looking for thank you.
If there is anything else i'll revert back to you
